src/main/java/org/springframework/data/repository/query/DefaultParameters.java

```diff
@@ -40,13 +42,30 @@ public DefaultParameters(Method method) {
 
 	/**
 	 * Creates a new {@link DefaultParameters} instance from the given {@link Method} and aggregate
-	 * {@link TypeInformation}.
+	 * {@link TypeInformation}. Note that this constructor uses a {@link IntFunction MethodParameterFactory} that doesn't
+	 * resolve generic method parameter types that are declared on the containing class. Use
+	 * {@link DefaultParameters#DefaultParameters(Method, IntFunction, TypeInformation)} with a
+	 * {@link MethodParameter#withContainingClass(Class) contextual factory} to ensure proper gerneric resolution.
 	 *
 	 * @param method must not be {@literal null}.
 	 * @param aggregateType must not be {@literal null}.
 	 */
 	public DefaultParameters(Method method, TypeInformation<?> aggregateType) {
-		super(method, param -> new Parameter(param, aggregateType));
+		this(method, index -> new MethodParameter(method, index), aggregateType);
```

Comment:
should we consider the aggregate type here and pass it on via `withContainingClass`?